index.js

```diff
@@ -140,16 +149,23 @@ tslintPlugin.report = function(reporter, options) {
             errorFiles.push(file);
             Array.prototype.push.apply(allFailures, failures);
 
-            if (reporter === 'json') {
-                jsonReporter(failures, file, options);
-            } else if (reporter === 'prose') {
-                proseReporter(failures, file, options);
-            } else if (reporter === 'verbose') {
-                verboseReporter(failures, file, options);
-            } else if (reporter === 'full') {
-                fullReporter(failures, file, options);
-            } else if (isFunction(reporter)) {
-                reporter(failures, file, options);
+            if (options.reportLimit >= 0 && options.reportLimit > totalReported) {
```

Comment:
By default reportLimit is -1. This PR breaks the reporters, as the if branch is never true if the reportLimit is -1. You can see this by running tests and comparing the results with the master brach, for example 'gulp invalid-prose' doesn't print the prose reporter in your feature branch.